build.gradle.kts

```diff
@@ -0,0 +1,71 @@
+import com.example.buildsrc.MyExtension
+
+plugins {
+    id("java")
+}
+
+group = "org.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnitPlatform()
+}
+
+allprojects{
+    extensions.create("Extension", MyExtension::class.java)
+
+    //passing a no-argument lambda makes the val initialized with lambda result
+    val addElementNoProjectAllScope by extra { project: Project ->
+        configure<MyExtension> {
+            contents.add("addElementNoProjectAllScope")
```

Comment:
Hmm why `...NoProject...`? after all, `project` is being passed to the lambda as a param explicitly

Reply:
But not used. I agree, it is quite misleading as the parameter is merely a placeholder so we get an actual lambda value.

Reply:
I assume the purpose of all these `contents` is just establish which (sub)project is being referenced by `configure`, right? Maybe there's an easier way to achieve this goal... idk, maybe just checking what's `this` within `configure<MyExtension> { ... }` block?